digest.go

```diff
@@ -101,26 +98,21 @@ func FromString(s string) Digest {
 // error if not.
 func (d Digest) Validate() error {
 	s := string(d)
-
 	i := strings.Index(s, ":")
-
-	// validate i then run through regexp
-	if i < 0 || i+1 == len(s) || !DigestRegexpAnchored.MatchString(s) {
+	if i <= 0 || i+1 == len(s) {
+		return ErrDigestInvalidFormat
+	}
+	algorithm, encoded := Algorithm(s[:i]), s[i+1:]
+	if algorithm == "" {
```

Comment:
You [can drop this block now, because the `i <= 0` check above covers it][1].

[1]: https://github.com/opencontainers/go-digest/pull/34#discussion_r117520102